plugins/jvm/ts/jvmHelpers.ts

```diff
@@ -100,9 +100,15 @@ namespace JVM {
     //must save to local storage, to be picked up by new tab
     saveConnection(clone);
     const $window: ng.IWindowService = HawtioCore.injector.get<ng.IWindowService>('$window');
-    let url = (clone.view || '/') + '?con=' + clone.name;
-    url = url.replace(/\?/g, "&");
-    url = url.replace(/&/, "?");
+    var url
+    if(clone.view) {
```

Comment:
I'm not sure what this is or how to test it, hence leaving it unchanged in a separate branch

Reply:
We prefer `let` or `const` over `var`. Also please add space between `if` and `(`.

Reply:
> Thanks, but can you tell what kind of issues this is going to fix?

My concrete example. I have a kubernetes ingress controller that routes /hawtio to hawtio. When opening a discovered connection, it looses the context path (/hawtio) and opens the new tab with root path (/) , hence opening the main application with hawtio query parameters.

Reply:
> We prefer `let` or `const` over `var`. Also please add space between `if` and `(`.

Have refactored to separate function to avoid using var set in several branches.